ui-modules/blueprint-composer/app/components/util/d3-blueprint.js

```diff
@@ -837,45 +850,50 @@ export function D3Blueprint(container, options) {
 
         _relationLabels.selectAll('.relation-label').remove(); // Re-draw labels every time, required to refresh changes.
 
-        // Group unique labels per path.
-        let labelsPerPath = {};
+        // Group unique relationships per path.
+        let relationshipsPerPath = {};
         _d3DataHolder.visible.relationships.forEach(d => {
             const key = getPathId(d);
-            if (!labelsPerPath[key]) {
-                labelsPerPath[key] = new Set();
+            if (!relationshipsPerPath[key]) {
+                relationshipsPerPath[key] = new Set();
             }
-            labelsPerPath[key].add(d.label);
+            relationshipsPerPath[key].add(getRelationId(d));
         });
 
+        const OFFSET_RATIO_PX = -12;
         const getLabelOffset = (d) => {
-            let labelIndex = Array.from(labelsPerPath[getPathId(d)]).indexOf(d.label);
-            return labelIndex > 0 ? labelIndex * -12 : 0;
+            let labelIndex = Array.from(relationshipsPerPath[getPathId(d)]).indexOf(getRelationId(d));
+            return labelIndex > 0 ? labelIndex * OFFSET_RATIO_PX : 0;
```

Comment:
Can be like so since the Set/array only contains string values.
```
  const hasLabel = Array.from(relationshipsPerPath[getPathId(d)]).includes(getRelationId(d));
  return hasLabel ? labelIndex * OFFSET_RATIO_PX : 0;
```

Reply:
`labelIndex` is required to calculate the offset.

Reply:
Good point, I can't read.